File: Service/Z2data.Core/Domain/Events/AffectedLocation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Z2data.Core.Domain.Lookups;
using Z2data.Core.Enums;

namespace Z2data.Core.Domain.Events
{
    public class AffectedLocation : BaseEntity
    {
        public object LocationId { get; set; }
        public string Name { get; set; }
        public Location Location { get; set; }
        public AffectedLocationType Type { get; set; }
        public IList<string> LocationType { get; set; }
        public string Iata { get; set; }
        public string Locode { get; set; }
        public Lookup[] DisasterTypes { get; set; }
        public double Distance { get; set; }
        public double? AssociatedDistance { get; set; }
        public Effect Status { get; set; }
        public bool IsConfirmed { get; set; }
    }
}

File: Front-end/src/app/events/meta-models/affected-location.ts
import { AffectedLocationType } from '../../shared/enums/affected-locationType';
import { Lookup } from '../../lookups/lookup';
import { Location } from '../../lookups/location';
import { Effect } from '../../shared/enums/effect';

export interface AffectedLocation {
    locationId: any;
    name: string;
    location: Location;
    type: AffectedLocationType;
    locationType: string;
    iata: string;
    locode: string;
    disasterTypes: Lookup[];
    distance: number;
    associatedDistance?: number;
    status: Effect;
    isConfirmed: boolean;
}
